src/main/webapp/view/discussion/discussion-manager.js

```diff
@@ -24,57 +24,66 @@ export default class DiscussionManager {
   static #PARAM_TIMESTAMP = 'timestamp';
   static #PARAM_TYPE = 'type';
   #lecture;
+  #displayedComments;
 
   /**
    * Creates a `DiscussionManager` to manage posting and fetching comments for
    * the `lecture`.
    */
   constructor(lecture) {
     this.#lecture = lecture;
+    this.#displayedComments = new Map();
   }
 
   /**
-   * Organizes comments into threads with nested replies.
-   *
-   * <p>All comments are sent from the servlet without any structure, so the
-   * client needs to organize them before displaying.
+   * Adds new comments to threads with nested replies by setting comment's
+   * parent and replies fields. Only processes and returns comments with ids
+   * that haven't already been seen. This is a private method and should only
+   * be called by `DiscussionManager`.
    */
-  structureComments(allComments) {
-    const commentIds = {};
+  processNewComments(allComments) {
+    const newComments = [];
+
+    // Comments are not guarenteed to have any specific order, so we need to do
+    // this loop before the other one to filter new comments and initialze
+    // fields.
     for (const comment of allComments) {
+      const id = comment.commentKey.id;
+      if (this.#displayedComments.has(id)) {
+        // This comment is already displayed.
```

Comment:
This comment doesn't seem necessary. I can tell from displayComments that it is already displayed.

Reply:
removed